dbt/include/spark/macros/materializations/incremental/strategies.sql

```diff
@@ -2,8 +2,14 @@
 
```

Comment:
Is this part working for you?

I tried using `insert_overwrite` on `iceberg` table and removing `partition_cols` didn't work for me and thus got SQL errors. I did some digging (without any prior knowledge with dbt) and I learned this from debugging:
target_relations seems to be created only from .sql statements and not populated from cache and thus it's is_iceberg,is_delta was always `None`

I am currently trying to find a fix, what seems to work is to change `target_relation` -> `existing_relation` as an argument for the function and also update the caller in `incremental.sql` file. Seems to work, but is it the right thing?

If it's working for you could you point me out to working example? Maybe I am doing something wrong.

My settup is basically:
```
{{ config(
  file_format='iceberg',
  materialized="incremental",
  incremental_strategy="insert_overwrite",
  partition_by=["part", "date(ts)"],
)}}


select
  *
  from {{source("catalog", "table")}}
  where {{ partitions_between(
          var('start_timestamp', current_timestamp()),
          var('end_timestamp', 'current_timestamp() + interval "1 hour"'))
        }}
  sort by part, ts
```

Reply:
I was able to reproduce the issue. Got it fixed by: https://github.com/dparent1/dbt-spark/pull/9

We already have the existing relationship, so I think it makes sense to pass that along. 

```sql
{{ config(
    file_format="iceberg",
    materialized="incremental",
    incremental_strategy="insert_overwrite",
    partition_by=["date(tpep_pickup_datetime)"]
) }}

SELECT *
FROM nyc.taxis
order by tpep_pickup_datetime
```

If we run it for the first time:
```sql
create or replace table default.taxis_partitioned
using iceberg
partitioned by (date(tpep_pickup_datetime))
as
SELECT *
FROM nyc.taxis
order by tpep_pickup_datetime
``` 

And the second run:
```sql
 insert overwrite default.taxis_partitioned
 select `VendorID`, `tpep_pickup_datetime`, `tpep_dropoff_datetime`, `passenger_count`, `trip_distance`, `RatecodeID`, `store_and_fwd_flag`, `PULocationID`, `DOLocationID`, `payment_type`, `fare_amount`, `extra`, `mta_tax`, `tip_amount`, `tolls_amount`, `improvement_surcharge`, `total_amount`, `congestion_surcharge`, `airport_fee` from taxis_partitioned__dbt_tmp
```

Which has the correct Iceberg syntax 👍🏻